astroid/rebuilder.py

```diff
@@ -119,10 +123,21 @@ def _get_doc(self, node: T_Doc) -> Tuple[T_Doc, Optional[str]]:
                 ):
                     doc = first_value.value if PY38_PLUS else first_value.s
                     node.body = node.body[1:]
-                    return node, doc
+                    if PY38_PLUS:
+                        doc_node = nodes.Const(
+                            value=doc,
+                            lineno=first_value.lineno,
+                            col_offset=first_value.col_offset,
+                            parent=node,
+                            end_lineno=first_value.end_lineno,
+                            end_col_offset=first_value.end_col_offset,
+                        )
+                    else:
+                        doc_node = None
```

Comment:
Even if the check might not really work for Python 3.7, why can't we create the node for it? It might be useful for some other check.

Reply:
I suggested against this because the line numbering of these nodes is (extremely) incorrect on `3.7`. See: https://github.com/PyCQA/astroid/pull/1276/commits/e61409f68ec297434a49ad83c7688facba301d99

`lineno` is actually `end_lineno`. Because of this I don't think it makes sense to make this node on `3.7` as the added benefit of `doc_node` over `doc` is incorrect and I suggested not adding it on `3.6` and `3.7` in https://github.com/PyCQA/astroid/pull/1276#issuecomment-986616113

Reply:
> I suggested against this because the line numbering of these nodes is (extremely) incorrect on `3.7`

Ok. Good point. Although it seems to work in some cases, eg.
```py
class X:
    """Some docstring."""
    var = 42
```

Multiline docstrings are an issue though.

--
Even with that, what I would like to prevent is having both `doc` and `doc_node` in parallel just because of it. It should just be one. If the line numbers are wrong, we should reset them manually. (Multiline docstrings are easy to identify as they set `col_offset = -1`.) In pylint you would then know to not rely on the line information. The docstring itself could still be accessed through `doc_node.value`.

I believe long term this is the better solution. Especially considering that Python 3.7 will also be deprecated at some point. We don't really need to talk about 3.6 anymore.

Reply:
Although I like that approach as it improves the information we provide to `pylint` and other users of `astroid` I wonder if this wouldn't create problems down the line.

What if we `add_message` on the docstring node and pass the node as `node` argument. That would trickle down to the final output and even the visual underlining in IDE's, as in: how should they handle a message with `col_offset = -1`. 
I wonder what the best place to address this issue is:
1) Make `pylint` check for "informational"/non-sensical `line_no` and `col_offset` (etc.) and handle them
2) Avoid `astroid` creating these "informational"/non-sensical attributes and handle them more gracefully here

I'm currently leaning towards the latter with a similar reasoning as to why I argued against setting `col_offset` to `0` in `pylint` when we don't have the correct value. Namely, I think we shouldn't add data that is not representative of the actual code and rather let users/IDE's handle missing data how they want to, ie. let them handle `col_offset = None`.

Reply:
IDEs already need to deal with both `col_offset = -1` and `col_offset = None`. There is actual a comment in the `vscode-python` code that mentioned this 😅 [vscode-python -> baseLinter.ts](https://github.com/microsoft/vscode-python/blob/44ef89560597ec319402b9361a37b8b344c04c41/src/client/linters/baseLinter.ts#L28-L30). So it wouldn't break anything (that isn't already broken).

> I wonder what the best place to address this issue is

My hope and what I know so far is that these issues are mostly fixed in newer versions. It would thus not be the biggest issue if it happen to be wrong, say in `3.6`.

In the end, I believe it's best if we can handle the remaining issues in astroid itself. In some cases a fix in pylint is just a lot quicker / easier.

For the particular issue here, my suggestion would be to set `col_offset = None` on those nodes. `end_lineno` and `end_col_offset` will already be `None` for 3.6 and 3.7.

Reply:
> IDEs already need to deal with both `col_offset = -1` and `col_offset = None`. There is actual a comment in the `vscode-python` code that mentioned this 😅 [vscode-python -> baseLinter.ts](https://github.com/microsoft/vscode-python/blob/44ef89560597ec319402b9361a37b8b344c04c41/src/client/linters/baseLinter.ts#L28-L30). So it wouldn't break anything (that isn't already broken).

😅 

> For the particular issue here, my suggestion would be to set `col_offset = None` on those nodes. `end_lineno` and `end_col_offset` will already be `None` for 3.6 and 3.7.

Agreed!

What about `lineno`? For multi-line docstrings it seems to be the actual `end_lineno`...

Reply:
> What about `lineno`? For multi-line docstrings it seems to be the actual `end_lineno`...

True, but we can't really determine a better one with complete certainty. At least that's still a valid `lineno` for the docstring. _The check probably be disabled for those versions anyway._

Reply:
So, the decision is what?

Reply:
> So, the decision is what?

Create the `doc_node` regardless of Python version. After that, we should take another look at the test output.
For the multiline docstring, we should probably do a conditional overwrite on 3.7 to replace `col_offset = -1` with `None`.

Reply:
done

Reply:
@DanielNoord @cdce8p so, what is now the plan for python3.7 and below?

Reply:
Create the `Const` as well and fix line numbers as we see fit (based on the test output). 

@cdce8p Already hinted towards one solution for multi line strings and their column number, but there might be other problems.

Reply:
done